components/server/src/iam/iam-session-app.ts

```diff
@@ -138,6 +142,20 @@ export class IamSessionApp {
         return existingUser;
     }
 
+    /**
+     * Updates `User.identities[current IdP].primaryEmail`
+     */
+    protected async updateOIDCUserOnSign(user: User, payload: OIDCCreateSessionPayload) {
+        const recent = this.mapOIDCProfileToIdentity(payload);
+        const existing = user.identities.find((identity) => identity.authId === recent.authId);
+
+        // Update email
+        if (existing && !!recent.primaryEmail && existing.primaryEmail !== recent.primaryEmail) {
+            existing.primaryEmail = recent.primaryEmail;
+            await this.userService.updateUserIdentity(user, existing);
+        }
+    }
```

Comment:
Could this whole method be moved into the `userService` such that it can be more easily re-used in other places? This would be particularly useful as we try to align with the service oriented architecture.

Reply:
Thought about that, but opted to not push the mapping in there. 

See, `userService.updateUserIdentity` does the actual update. On top, here is the mapping of the profiles, which very well could be pulled out to be reusable.